Fix typos and capitalization of "Unicode"

waldyrious opened the pull request:
- Consistently capitalize the word "Unicode" across the codebase
- Fix several typos

jonathandturner commented:
Looks like you need to update the `no_discarded_white_space_start_of_comment` test as well

waldyrious commented:
@jonathandturner Thanks for the heads-up. I've fixed that test, added a couple more typo fixes, and an extra commit cleaning up mixed whitespace in `crates/nu-parser/src/lex/tests.rs`.

jonathandturner commented:
Looks good!
